content/tutorials/git_authentication/index.Rmd

```diff
@@ -0,0 +1,266 @@
+
+1. Turn on two-factor authentication for your GitHub account.
+1. Adopt HTTPS as your Git transport protocol.
+1. Use a personal access token (PAT) for all Git remote operations from the command line or from R.
+1. Allow tools to store and retrieve your credentials from the Git Credential Manager.
+
+This deserves some further explanation.
+The first recommendation essentially is an extra layer of security compared to a simple username and password only authentication.
+To enable two-factor authentication for your GitHub account, follow [these steps](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).
+Another two-factor authentication app that is not mentioned on that website but easy to use is Google Authenticator.
+
+The second recommendation is to use the [HTTPS protocol](#https) to transport information (data) from your local Git repositories to their remote (online) counterparts.
+A good alternatives is the SSH protocol. Do not use the HTTP protocol as it is not secure.
+We have also included in this tutorial how to use the [SSH protocol](#ssh), but in our experience this is not necessary in most circumstances and more involved to implement in Windows.
+Users of the Linux operating system, on the other hand, may prefer the SSH protocol.
+It is always possible to switch between HTTPS and SSH protocols and we explain in that section how to do that.
+
+The third recommendation is mandatory when you use the HTTPS protocol.
+GitHub no longer allows a simple password for Git remote operations.
+The remote is the place where you store your code on a Git Server, for instance https://github.com/inbo/tutorials.
+Git remote operations include, among others, `git clone`, `git fetch`, `git pull` and `git push`.
+
+The fourth recommendation is the easiest, because it just works out of the box with recent versions of Git (2.29 or higher).
+The installation of Git on Windows comes with a [Git Credential Manager](https://github.com/GitCredentialManager/git-credential-manager) which allows that web apps like [hackmd](https://hackmd.io/login) or R packages like `gert` and `gh` can ask for your credentials and get them from the credential store.
+In earlier days, it was sometimes needed to store your PAT in a `.Renviron` text file which you store only locally (the file will in that case contain a line GITHUB_PAT=...).
+This is no longer needed and should be avoided because of the risk to expose the PAT when that file is accidentally put online.
+You can check if you have previously set a GITHUB_PAT in your `.Renviron` with the R command `usethis::edit_r_environ()`.
+
+## Creating Personal Access Tokens (PAT)
+
+We move the steps about creating a PAT upfront, because this includes guidelines to store the PAT to the Git Credential Manager so they can be discovered automatically.
```

Comment:
```suggestion
We move the steps about creating a PAT upfront, because this includes guidelines to store the PAT with the Git Credential Manager so they can be discovered automatically.
```